src/base/bittorrent/trackerentry.cpp

```diff
@@ -28,22 +28,16 @@
 
 #include "trackerentry.h"
 
-#include <QUrl>
-
 bool BitTorrent::operator==(const TrackerEntry &left, const TrackerEntry &right)
 {
-    return ((left.tier == right.tier)
-        && QUrl(left.url) == QUrl(right.url));
+    return (left.url == right.url);
```

Comment:
IMO for official `operator==` and `qHash`, their definition looks strange... 
Without regard to its usage, is it correct to prove equivalence based on solely the `url` field?

Reply:
> Without regard to its usage, is it correct to prove equivalence based on solely the `url` field?

Yes. Because libtorrent doesn't really support entries with the same `url` (existing inconsistencies are considered as a bug that should be fixed soon).


> IMO for official `operator==` and `qHash`, their definition looks strange...

I agree. But they have a purely utilitarian meaning in order to filter out entries with the same `url` through QSet/QHash. Maybe it's worth declaring them privately in a `torrentimpl.cpp` file (if possible)?

Reply:
I even think about making TrackerEntries to be Identities (see [Identity vs Value](https://doc.qt.io/qt-5/object.html#qt-objects-identity-vs-value) for example) , since they look like such in their essence. But not in this PR.

Reply:
>Maybe it's worth declaring them privately in a torrentimpl.cpp file (if possible)?

I think it is feasible. To be precise, I'm referring to `operator==()`, `operator<()`, `qHash()`.
And perhaps only the implementation should be moved to torrentimpl.cpp while the function prototype can still stay in trackerentry.h?
Or we can have a generic definition in trackerentry.cpp and let it be override by specific usage in each .cpp file?

Reply:
> And perhaps only the implementation should be moved to torrentimpl.cpp while the function prototype can still stay in trackerentry.h?
> Or we can have a generic definition in trackerentry.cpp and let it be override by specific usage in each .cpp file?

I still think TrackerEntry shouldn't be generally comparable.
I'm going to finish this PR and take a little break from this topic. I hope I'll find a more elegant solution later.

Reply:
>I'm going to finish this PR and take a little break from this topic. I hope I'll find a more elegant solution later.

OK.